modules/bitcoind.nix

```diff
@@ -386,7 +388,12 @@ in {
           ''
         ) (builtins.attrNames cfg.rpc.users);
       in ''
-        ${optionalString cfg.dataDirReadableByGroup "chmod -R g+rX '${cfg.dataDir}/blocks'"}
+        ${optionalString cfg.dataDirReadableByGroup ''
+          if [[ -e '${cfg.dataDir}/blocks' ]]; then
+            chmod -R g+rX '${cfg.dataDir}/blocks'
```

Comment:
If this bitcoind service starts for the first time, the blocks dir does not exist yet, so `cfg.dataDirReadableByGroup` won't have the desired effect?

Reply:
With `dataDirReadableByGroup = true`, the blocks dir is automatically created and populated with group read permissions.

See:
```bash
# dataDirReadableByGroup = false
run-tests.sh -s bitcoind container --run c ls -al /var/lib/bitcoind
# Result
drwx------ 3 bitcoin bitcoin 4096 Feb  5 21:45 blocks

# dataDirReadableByGroup = true
run-tests.sh -s "{
  services.bitcoind.enable = true;
  services.bitcoind.dataDirReadableByGroup = true;
}" container --run c ls -al /var/lib/bitcoind
# Result
drwxr-x--- 3 bitcoin bitcoin 4096 Feb  5 21:46 blocks
```

`chmod -R g+rX '${cfg.dataDir}/blocks'` only exists so that users can enable `dataDirReadableByGroup` on an existing bitcoind instance that was created without `dataDirReadableByGroup`.

Reply:
Ah sorry, I missed the umask setting.